lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -182,25 +182,29 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
         ) ::
           {:ok, list(SszTypes.validator_index())} | {:error, binary()}
   def compute_committee(indices, seed, index, count) do
-    start_ = div(length(indices) * index, count)
-    end_ = div(length(indices) * (index + 1), count) - 1
-
-    case compute_committee_indices(start_, end_, indices, seed) do
-      {:ok, result_list} -> {:ok, Enum.reverse(result_list)}
-      _ -> {:error, "invalid index_count"}
-    end
-  end
-
-  defp compute_committee_indices(start_, end_, indices, seed) do
-    Enum.reduce_while(start_..end_, {:ok, []}, fn i, {:ok, acc_list} ->
-      case compute_shuffled_index(i, length(indices), seed) do
-        {:ok, shuffled_index} ->
-          {:cont, {:ok, [Enum.at(indices, shuffled_index) | acc_list]}}
-
-        {:error, _} = error ->
-          {:halt, error}
-      end
-    end)
+    index_count = length(indices)
+    a = div(index_count * index, count)
+    b = div(index_count * (index + 1), count) - 1
+
+    to_swap_indices =
+      a..b//1
+      |> Stream.map(&compute_shuffled_index(&1, index_count, seed))
+      |> Stream.map(fn {:ok, i} -> i end)
```

Comment:
What if this is not `:ok`? We should fix that edge case. E.g. with a reduce similar to what we use in SSZ when we have to encode several types and they fail.

Reply:
Done